src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventLogger.cs

```diff
@@ -6,31 +6,47 @@
 using System.Linq;
 using Microsoft.Azure.WebJobs.Script.Config;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 
 namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
 {
-    public class DiagnosticEventLogger : ILogger
+    public class DiagnosticEventLogger : ILogger, IDisposable
     {
         private readonly IDiagnosticEventRepositoryFactory _diagnosticEventRepositoryFactory;
         private readonly IEnvironment _environment;
+        private readonly IDisposable _standbyChangeHandler;
+
         private IDiagnosticEventRepository _diagnosticEventRepository;
+        private bool _isEnabled;
 
-        public DiagnosticEventLogger(IDiagnosticEventRepositoryFactory diagnosticEventRepositoryFactory, IEnvironment environment)
+        public DiagnosticEventLogger(IDiagnosticEventRepositoryFactory diagnosticEventRepositoryFactory, IEnvironment environment,
+            IOptionsMonitor<StandbyOptions> standbyOptions)
         {
             _diagnosticEventRepositoryFactory = diagnosticEventRepositoryFactory;
             _environment = environment;
+
+            SetEnabledState(standbyOptions.CurrentValue);
+
+            // Re-evaluate whether this is enabled after specialization
+            if (standbyOptions.CurrentValue.InStandbyMode)
```

Comment:
This seems like a perfect use of options `OnChange`. However, you could alternatively just evaluate this within `IsEnabled()`:

``` CSharp
public bool IsEnabled(LogLevel logLevel) => !_options.CurrentValue.InStandbyMode &&
    !FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagDisableDiagnosticEventLogging, _environment);
```

Reply:
That was my initial approach, but here's what I went through in my head:
- `FeatureFlags.IsEnabled()` does some string parsing -- which I wanted to avoid doing on every log message.
- So I started trying to refactor `FeatureFlags` so it cached internally in a `HashSet<string>` for fast lookup and no parsing.
- But `IEnvironment` values can change (tests can do it; specialization can do it), which means that we'd need a way to also store/compare that original string and invalidate the cache if it ever changed.
- You can also technically pass in different `IEnviroment` objects on every call -- so we'd need to maintain a cache for each one I'd imagine.
- Would we need to introduce locks to make sure it was thread-safe? It looks like we only access this in about a dozen places, so maybe I could see if it mattered to any of them. (note we've had a *ton* of threading issues during specialization in the past so we're very cautious about stuff changing mid-specialization).

Then I felt like I was way too into the weeds and went back to a pattern we already use elsewhere.